### PR TITLE
docs: removed show more from code example

### LINKED ISSUE #3242: [Feature Request] Documentation: Remove the "show more" button in the component's code example sections
### Is your feature request related to a problem? Please describe.

As I am starting a new project and use NextUI for the first time, I use the documentation extensively and want to navigate fast to code examples. Right now, each code example is 2 clicks away, first switching tabs from "preview" to "code". Then there appears an uninformative and none interactive code excerpt with only the imports readable and the rest faded out, with a "show more" button. Only after clicking this, I can see useful examples of the component usage. 

Is there a rational behind this extra two-step interaction to access the examples? Is it to prevent bots scraping the code?

### Describe the solution you'd like

Could we consider removing this extra "show more" button? It would much improve the usability of the documentation in my opinion. 

### Describe alternatives you've considered

Documentation at mui.org sometimes shows relevant short code examples for components directly under the preview panes, with the option to open extended code listings with one click.

Documentation at tailwindui.com also shows full code examples directly after switching tabs between "preview" and "code".

### Screenshots or Videos

_No response_

### PR DESCRIPTION
**Pull request by ankitts**
Closes #3242 

## 📝 Description

This PR removes the additional step of clicking on Show More while viewing a code example in docs.

## ⛳️ Current behavior (updates)

Currently, to view a code example from docs fully, we need to click on the show more button.
Screenshot 2024-06-30 145407

## 🚀 New behavior

The code can be seen fully after switching to preview tab without the need to click one additional button.
Screenshot 2024-06-30 145331

## 💣 Is this a breaking change (Yes/No):

No

## Summary by CodeRabbit

- **New Features**
  - Simplified the `CodeDemo` and `Sandpack` components by removing the `defaultExpanded` prop.
  - Improved the `SandpackCodeViewer` component by removing unnecessary state management and simplifying logic.

- **Style**
  - Updated `sandpack.css` to remove outdated styles and adjust scrollbar visibility.

- **Refactor**
  - Removed unused imports and redundant code from `SandpackCodeViewer` component.

### CODE REVIEW

**Comment by wingkwong** [comment]
@ankitts This issue is assigned to our team member. Next time please ask for an assignment before working on it. This would save both of our time and avoid duplicate work.

**Review by winchesHe — COMMENTED** [review]
@ankitts Thx you for your working, I left some comments

**Comment by winchesHe** [comment]
`apps/docs/styles/sandpack.css`

It can be deleted
```css
.sp-code-viewer.is-expanded .cm-scroller {
  overflow: auto;
  padding-bottom: 50px;
}
```

File: apps/docs/components/sandpack/code-viewer.tsx [hunk 1]
```diff
@@ -33,17 +31,15 @@ export interface CodeViewerProps {
...
     const {sandpack} = useSandpack();
     const {code} = useActiveCode();
 
     const {activeFile} = sandpack;
 
-    const [isExpanded, setIsExpanded] = React.useState(defaultExpanded);
-
     // const id = React.useId();
     // hack to make sure we re-render the code editor and change current file
     // TODO: open an issue on sandpack-react
```
Comment (winchesHe) [inline apps/docs/components/sandpack/code-viewer.tsx:45]: Line 47-51 can be commented

```ts
const lineCountRef = React.useRef<{[key: string]: number}>({});
    if (!lineCountRef.current[activeFile]) {
      lineCountRef.current[activeFile] = code.split("\n").length;
    }
```

**Comment by ankitts** [comment]
> @ankitts Thx you for your working, I left some comments

Thank you for the comments :) I have made the said changes.

**Comment by winchesHe** [comment]
@wingkwong Could you help agree to the deployment document🫰

**Review by wingkwong — APPROVED** [review]

**Review by jrgarciadev — APPROVED** [review]

**Pull request merged**